### PR TITLE
[WIP] Polymer 1.0

### PR DESCRIPTION
**Pull request by arfon**
Nothing much to see here so far. These changes are simply the result of running: `find . -name "*.html" -execdir polyup --overwrite "{}" \;` in `app/assets/components`.

### CODE REVIEW

File: app/assets/components/elements/json-delete.html [hunk 1]
```diff
@@ -1,9 +1,18 @@
...
-    <script>
-        Polymer({
-            method: 'delete'
-        });
-    </script>
-
-</polymer-element>
+<!--
+    TODO(polyup): Inheriting from other custom elements is not yet supported.
+    See: https://www.polymer-project.org/1.0/docs/migration.html#inheritance
```
Comment (arfon) [inline app/assets/components/elements/json-delete.html]: @stuartlynn @marcrohloff it looks like we're going to need to change some of the element inheritance stuff here.

Comment (stuartlynn) [inline app/assets/components/elements/json-delete.html]: Yeah. Its a fairly big change. Happy to work on it but we should figure out
a plan of attack.

There is a tool called polyup that can automate many of the changes here :
https://www.polymer-project.org/1.0/docs/migration.html
Also a good guide here
http://jcrowther.io/2015/06/01/upgrading-from-polymer-v0-5-v1-0/

Might be worth a quick call to plan how we are going to do this?

Cheers

Stuart

On Thu, Jan 7, 2016 at 9:45 AM Arfon Smith <notifications@github.com> wrote:

> In app/assets/components/elements/json-delete.html
> <https://github.com/openjournals/theoj/pull/256#discussion_r49079222>:
>
> > @@ -1,9 +1,18 @@
> > -<polymer-element name=json-delete extends=json-req>
> > -
> > -    <script>
> > -        Polymer({
> > -            method: 'delete'
> > -        });
> > -    </script>
> > -
> > -</polymer-element>
> > +<!--
> > +    TODO(polyup): Inheriting from other custom elements is not yet supported.
> > +    See: https://www.polymer-project.org/1.0/docs/migration.html#inheritance
>
> @stuartlynn <https://github.com/stuartlynn> @marcrohloff
> <https://github.com/marcrohloff> it looks like we're going to need to
> change some of the element inheritance stuff here.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openjournals/theoj/pull/256/files#r49079222>.
>
-- 
*Stuart Lynn*
Map Scientist

*CartoDB—Map Your World's Data*
143 Roebling St 2A, New York, 11211, USA
T: +3123940797  Twitter: @stuart_lynn
W: www.cartodb.com

Live sessions to learn CartoDB

Comment (marcrohloff) [inline app/assets/components/elements/json-delete.html]: I can try and help some next week.

Comment (marcrohloff) [inline app/assets/components/elements/json-delete.html]: I assume you can replace inheritance with mix-ins (or whatever polymer calls them)

**Comment by arfon** [comment]
@stuartlynn - this PR is the result of running the Polyup tool on the current code. It looks to have left (reasonably?) useful notes about what is likely to be broken at this stage.

Not sure if you're planning on looking at any of this over the weekend but I thought I'd let you know that Polyup has already done its stuff on this PR :smile:

**Comment by stuartlynn** [comment]
My plan was to run the polyup tool and go from there so thats cool thanks.
Will let you know how far I get.

Cheers


On Sat, Jan 9, 2016 at 11:14 AM Arfon Smith <notifications@github.com>
wrote:

> @stuartlynn <https://github.com/stuartlynn> - this PR is the result of
> running the Polyup tool on the current code. It looks to have left
> (reasonably?) useful notes about what is likely to be broken at this stage.
>
> Not sure if you're planning on looking at any of this over the weekend but
> I thought I'd let you know that Polyup has already done its stuff on this
> PR [image: :smile:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openjournals/theoj/pull/256#issuecomment-170256915>.
>
-- 
*Stuart Lynn*
Map Scientist

*CartoDB—Map Your World's Data*
143 Roebling St 2A, New York, 11211, USA
T: +3123940797  Twitter: @stuart_lynn
W: www.cartodb.com

Live sessions to learn CartoDB

**Comment by stuartlynn** [comment]
Ok so updated and resolved a few issues. Updated the polymer library and webcomponents, removed extra include tags that polyup included and have fixed a few bugs it introduced. Getting a lot of errors from the old polymer components. I think we are going to have to replace all the polymer components we used with their polymer 1.0 counterparts. 

@marcrohloff  where we handling the component installs with bower? I cant quiet remember. If so we might just be able to call update on a bunch of them but we might have to switch out a bunch by hand for the newer versions. 

So for example we need to upgrade from core-ajax to iron-ajax  https://elements.polymer-project.org/elements/iron-ajax . Not sure if polyup does this automatically when it finds import statements and we just don't have those because of the asset pipeline.

If we are using bower some of that might be possible to upgrade as follows https://github.com/PolymerLabs/polyup/blob/master/AfterUseGuide.md#bower

I cant see a bower file in the root dir and I cant remember how we where handling the dependancies.

**Comment by marcrohloff** [comment]
I wasn't using bower, I don't know if that has changed.

**Comment by stuartlynn** [comment]
Ok I am going to start swapping out components tonight. I might set up bower to install to the vendor directory and set up a bower.json file for the components so its easy to upgrade in future. Let me know if there are any objections

**Comment by arfon** [comment]
> Ok I am going to start swapping out components tonight. I might set up bower to install to the vendor directory and set up a bower.json file for the components so its easy to upgrade in future. Let me know if there are any objections

Sounds good to me.

**Comment by arfon** [comment]
@stuartlynn - did you make any further progress on this? I believe @marcrohloff is going to try and chip in this week so it would be good to get a status update from you.

File: app/assets/components/ui-elements/oj-review-paper-info.html [hunk 2]
```diff
@@ -129,6 +129,7 @@
               font-weight: 700;
               /*line-height: 30px;/* Approximation due to font substitution */
               max-height:  92px;
+              overflow-y:  hidden;
```
Comment (arfon) [inline app/assets/components/ui-elements/oj-review-paper-info.html]: I think this is probably better than `scroll` but would it be possible to instead truncate the title? Is there a Polymer helper function to do this?

Comment (arfon) [inline app/assets/components/ui-elements/oj-review-paper-info.html]: Or would `text-overflow: ellipsis;` work here?

**Comment by marcrohloff** [comment]
wrt to the title overflow/ellipses. The `overflow:ellipses` options does not, unfortunately, work with multiline text only with single line. 
There is nothing in Polymer to do the ellipses. You can truncate at `n` characters but that is tricky with non-fixed width fonts. There are also some jQuery add-ins that do the ellipses. I have just found them to be unreliable. Probably because I have my screen scaled.
The last option I've seen is to fade out the last line.

https://css-tricks.com/line-clampin/

**Comment by marcrohloff** [comment]
I'll merge this for now. Just so that we are all clear - we are now working off the `master` branch.

**Comment by marcrohloff** [comment]
Also I'm creating a `polymer-0.5` branch with the old code for posterity aka reference

**Pull request merged**